### PR TITLE
feat(version): Add GITHUB_SHA information to builds

### LINKED ISSUE #305: Craft should report its version more accurately
Currently, Craft keeps reporting the last released version for builds from master. Moreover it lacks the git SHA that it was built from. These make debugging and bisecting harder when things go wrong.

Craft should report a proper "dev" version along with the git SHA that it was based on.

### PR DESCRIPTION
**Pull request by BYK**
This patch adds GITHUB_SHA to version information for better
debugging. It also replaces the bump-version script with a
short node call, and adds a post release script that bumps
the version to a pre-minor with the `dev` suffix.

Fixes #305.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@ jobs:
         if: steps.cache.outputs.cache-hit != 'true'
         run: yarn install --frozen-lockfile
       - name: Build
-        run: yarn build
+        run: yarn build --define:process.env.CRAFT_BUILD_SHA='"'${{ github.sha }}'"'
```
Comment (chadwhitacre) [inline .github/workflows/build.yml:48]: That's some fancy quoting. 😳

Comment (BYK) [reply .github/workflows/build.yml:48]: 🤷🏻‍♂️

**Comment by BYK** [comment]
Okay, verified both artifacts:

1. Downloaded the [build artifacts from GitHub](https://github.com/getsentry/craft/suites/3955021823/artifacts/99287212) and ran the built binary with `--version`
2. Checked [GCB output](https://console.cloud.google.com/cloud-build/builds/f4f11e3f-5078-4aa4-af56-5936d9891302;step=3?authuser=1&project=sentryio) to verify correct version information (line 37)

**Review by chadwhitacre — APPROVED** [review]
```
/Users/chadwhitacre/Desktop/foo/dist
total 2992
-rwxr-xr-x@ 1 chadwhitacre  staff  3062227 Oct  4  2021 craft*
$ ./craft --version
0.26.0-dev.0 (fd7e626daa15d6a7e1387f5278ca42e357fee216)
$
```

**Pull request merged**